src/renderer/domains/network/registry/service/chainRegistry.ts

```diff
@@ -0,0 +1,133 @@
+import mitt, { type Emitter } from 'mitt';
+import { type PolkadotClient, createClient } from 'polkadot-api';
+import { type WsEvent, type WsJsonRpcProvider, getWsProvider } from 'polkadot-api/ws-provider/web';
+
+import { chainsService } from '@/shared/api/network';
+import { type Chain, type ChainId } from '@/shared/core';
+import { nullable } from '@/shared/lib/utils';
+import { CONFIG } from '../lib/constants';
+import { type ChainApi } from '../lib/types';
+
+type RegistryEvents = {
+  status: {
+    chainId: ChainId;
+    type: WsEvent;
+  };
+};
+
+class ChainRegistry {
+  // TODO: support Light Clients in future
+  #storage: Map<
+    ChainId,
+    {
+      provider: WsJsonRpcProvider;
+      client: PolkadotClient;
+    }
+  > = new Map();
+
+  #subscribers: Map<ChainId, VoidFunction[]> = new Map();
```

Comment:
I know it's a matter of taste but isn't it slightly easier to read?

```suggestion
  #storage = new Map<
    ChainId,
    {
      provider: WsJsonRpcProvider;
      client: PolkadotClient;
    }
  >();

  #subscribers = new Map<ChainId, VoidFunction[]>();
```